### DISCUSSION TITLE
How do I disable kubernetes auto-discovery?

### DISCUSSION THREAD

**Discussion by PeterGrace** [gravitational/teleport#48867] [Q&A] [answered]
When I added my EKS cluster to the teleport instance I neglected to notice it having "auto-discovery" enabled.  Due to the workload in this cluster, I now have hundreds of services listed that I didn't intend to create!

I tried editing the configmap on the destination cluster and re-starting the teleport agent, however when it came back up the configmap was reverted to 'enabled: true'

If possible, can you link me where in the public docs this is answered?  I feel like there's not a lot of discoverable documentation on managing resources after they are created.

Thanks!

**Comment by stevenGravy** [comment, marked as answer]
When you configure the roles in the values.yaml update to only `kube`. Be default it includes `kube,app,discovery`.

```yaml
roles: kube
```

**Reply by PeterGrace** [reply]
Thanks for your reply @stevenGravy , however I installed the teleport agent via the EKS IAM-API method as opposed to deploying via a helm chart.  That's why I figured editing the configmap should work, but it seems like maybe the teleport instance uses some other method to deploy via that eks-iam path that necessitates me editing a config closer to teleport?

**Reply by stevenGravy** [reply]
When this is setup there is an option to not have app discovery. I would take a look at the `tctl get discovery_config` on what's set.

![image](https://github.com/user-attachments/assets/f31b0ade-f844-4e6c-ae38-f921d0498092)

**Reply by PeterGrace** [reply]
@stevenGravy 
```
  $   kubectl exec -ti deployment/teleport-auth -- tctl get discovery_config
[]
```

Using your hint here about `discovery_config` I see there's a page about [configuring application discovery](https://goteleport.com/docs/reference/agent-services/kubernetes-application-discovery/) which seems to indicate I should be able to edit the configmap to turn it off, but as I mention when I edit it, it reverted back to `enabled: true` within a few seconds.

**Reply by stevenGravy** [reply]
If you did a individual add I would just remove that kubeagent install in the namespace.

**Reply by PeterGrace** [reply]
> If you did a individual add I would just remove that kubeagent install in the namespace.

I don't want to remove access to the cluster, I just want to disable autodiscovery.  Is this not possible?

**Reply by stevenGravy** [reply]
You should be able to set the Teleport config to turn off discovery and app service. You have to edit in the kube agent config map.

**Reply by PeterGrace** [reply]
Thanks @stevenGravy, It was, as you said, that `roles` setting in the helm values.  I had forgotten that I'd added the teleport agent to our argocd instance (it was a long day yesterday) and so every time I tried to edit the configmap, argocd was changing it back.
